druid/src/text/storage.rs

```diff
@@ -45,6 +54,24 @@ pub trait TextStorage: PietTextStorage + Data {
     }
 }
 
+/// Provides information about keys change for more fine grained invalidation
+pub struct EnvUpdateCtx<'a, 'b>(&'a UpdateCtx<'a, 'b>);
```

Comment:
Is there a reason not to just put these methods on `UpdateCtx` directly?

Reply:
Because `UpdateCtx` has lot of methods like `set_cursor` we don't want `TextStorage` to use.

Reply:
Okay. the `storage` mod isn't public, so you might want to reexport this from `druid::text`?

Reply:
it is reexported already

Reply:
My work here is done 🤝